lib/api/index.js

```diff
@@ -183,6 +186,32 @@ app.post('/:installationId/sync', bodyParser, async (req, res) => {
   }
 })
 
+app.post('/jira/:installationId/verify', bodyParser, async (req, response) => {
+  const { installationId } = req.params
+  const installation = await Installation.findById(installationId)
+
+  const respondWith = function (message) {
+    const data = {
+      message: message,
+      installation: { enabled: installation.enabled, id: installation.id, jiraHost: installation.jiraHost }
+    }
+
+    return response.send(JSON.stringify(data))
+  }
+
+  if (installation.enabled) {
+    respondWith('Installation already enabled')
+  } else {
+    await verifyInstallation(installation, req.log)()
+
+    if (installation.enabled) {
```

Comment:
Am I right, that the idea here is, that the function `respondWith` closes (as in closure) over the mutable `installation` and that `verifyInstallation` mutates `installation`?
If so, I am a little afraid of this shared mutable state.

I personally would have loved to have `verifyInstallation` to return a new object instead of mutating its argument.

Reply:
Your understand of this code is bang on.

I like the idea of returning a mutated copy but I'm unsure if it fits this context. `installation` is a Sequelize model that behaves like ActiveRecord: to change the row, you update the object's attribute then persist that in the database. We want to set `enabled=true`. To avoid mutation, we would need to do something like this:

```javascript
const verifyInstallation = async function (installation) {
  const installationCopy = installation.copy() // Hand-waving -- not sure if Sequelize supports this
  // ...verify the install
  await installationCopy.update({enabled: true})
  return installationCopy
}
```

I think we could make this work but I'm hesitant to go against the conventions of Sequelize.

**Disclaimer**: I'm new to Node and Sequelize so I may be missing something obvious. If so, please tell me! 😄

Reply:
Got your point. What about just returning the mutated installation from that function so that the code is more obvious. Especially the second (nested) `if (installation.enabled)`.
Even though technically it is the same object, the API would be cleaner and would allow potential future changes, even though we stick with the current semantics for now.
I agree however that we stick with conventions (internally). But I am also not a Sequelize expert ✌️

Reply:
I need to refresh more—just saw this comment. I've made a note of this and will make the change along with the tests.